### PR TITLE
Fix unwinnable game situations in RA-Allies05a

### PR DESCRIPTION
**Pull request by Muh-Muh**
A determined player could infiltrate the wrong war factory or none at all. Both led to an unwinnable game. For ingame testing [if that is even required] I suggest ample use of /kill or a good micro against V2's.

### CODE REVIEW

**Comment by MustaphaTR** [comment]
Considering Targetable is conditional, i would prefer making other WFs uninfiltrateable using conditions given by lua.

**Comment by Muh-Muh** [comment]
Updated.
I feel like there is no good reason to prevent the player from infiltrating the prison directly. Imho the player shouldn't be punished for, erm.., creativity. So I left that part in.
Instead of toggling the target types in lua I simply made a new war factory that can't be targeted by the spy.
I'm still not quite sure how all the YAML traits and definitions work, so perhaps there is a shorter way to achieve that. If so, or if I should make the prison only targetable after the war factory has been infiltrated let me know.

File: mods/ra/maps/allies-05a/allies05a.lua [hunk 1]
```diff
@@ -240,6 +240,9 @@ InitTriggers = function()
 	end)
 
 	Trigger.OnInfiltrated(Prison, function()
+		if not greece.IsObjectiveCompleted(infWarfactory) then
+			Media.DisplayMessage("Good work! But next time skip the heroics!", "Battlefield Control")
+			greece.MarkCompletedObjective(infWarfactory) end
```
Comment (abcdefg30) [inline mods/ra/maps/allies-05a/allies05a.lua]: Can you put the `end` on it's own line, please?

**Review by abcdefg30 — COMMENTED** [review]
Agree that we don't need to lock the prison until the warfactory is captured. It would be bad if a player found a way to sneak around everything just to discover that they have to go back to infiltrate the factory...

File: mods/ra/maps/allies-05a/allies05a.lua (hunk 1 above)
Comment (Muh-Muh) [reply mods/ra/maps/allies-05a/allies05a.lua]: Done.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(Next-Playtest)/_compare/5de5b653fb2f33be0b4000fb70a19399dd02243a...c37cb8fcbfb7d555e2ff55c9886653c497b1349a)

Added to `prep-1801` as discussed on IRC:
https://github.com/OpenRA/OpenRA/commit/3c87adbad55014178be6f479b1d2e1761df704cc
https://github.com/OpenRA/OpenRA/commit/16e63a948767a4e5908e957912f6bb9160b31b08

**Pull request merged**